### PR TITLE
healthcare API: change how HL7v2MessageList processes results, fix er…

### PR DESCRIPTION
**Pull request by noerog**
…ror in a path name in HL7v2MessageGet

Fixes:

b/156347655
b/156336144
b/156395753

### CODE REVIEW

File: healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageList.java [hunk 1]
```diff
@@ -63,21 +59,21 @@ public static void hl7v2MessageList(String hl7v2StoreName) throws IOException {
...
+          continue;
+        }
+        System.out.printf("Retrieved %s HL7v2 messages: \n", response.getHl7V2Messages().size());
+        for (Message message : response.getHl7V2Messages()) {
+            System.out.println("\t" + message);
+        }
+        request.setPageToken(response.getNextPageToken());
+       } while (response.getNextPageToken() != null);
+    } catch (IOException e) {
+      System.out.println("Unable to list HL7v2 messages: \n" + e.toString());
```
Comment (lesv) [inline healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageList.java:76]: You don't need the extra `\n` when using println

File: healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageGet.java [hunk 2]
```diff
@@ -32,7 +31,7 @@
 
 public class HL7v2MessageGet {
   private static final String MESSAGE_NAME =
-      "projects/%s/locations/%s/datasets/%s/hl7V2Store/%s/messages/%s";
+      "projects/%s/locations/%s/datasets/%s/hl7V2Stores/%s/messages/%s";
```
Comment (lesv) [inline healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageGet.java:34]: How did this ever work?

File: healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageList.java (hunk 1 above)
Comment (noerog) [reply healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageList.java]: Done.

File: healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageGet.java (hunk 2 above)
Comment (noerog) [reply healthcare/v1/src/main/java/snippets/healthcare/hl7v2/messages/HL7v2MessageGet.java:34]: Most of the fixes I've submitted in the past few days are due to the tests being correct and passing but some issue in the sample itself causing it to fail. We use a consistent "projects/%s/locations/%s/datasets/%s/hl7V2Stores/%s/messages/%s" string in tests so it always passes there.

**Comment by noerog** [comment]
Everything passing now. Please merge. Thanks.

**Pull request merged**